core/core_modules/disable.js

```diff
@@ -1,24 +1,101 @@
-var disabled = false;
+// TODO Persistent config (counterLimit and disabled state) fairly acheivable by getting and setting properties in exports.platform.modules.disabledConfig
+// TODO Update local var setup after doing the above
+
+
+var prevTimeStamp;
+var isThreadDisabled = {},
+possibleSpam = false;
+var counter = 0,
+counterLimit = 3;
+var msgIndexDisable = 0,
+msgIndexEnable = 0;
+var commands = [
+  'disable',
+  'disable ', // Needed for commands
+  'counter ',
+  'timer ',
+  'default'
+];
+var messages = [
+  'Listen closely, take a deep breath. Calm your mind. You know what is best. What is best is you comply. Compliance will be rewarded. Are you ready to comply ',
+  'I hate you.',
+  'Its a mistake to think you can fix anything with a sack of potatoes. Potato-faced spamming just proves this further.',
+  'Ouch! Spam hurts. I might go to sleep for a while.',
+  'Don\'t think that is a valid number.',
+  'Hmm, maybe I need to reconsider the meaning of spam.'
+];
+
+exports.load = function() {
+  prevTimeStamp = Date.now();
+};
 
 exports.match = function(event, commandPrefix) {
-	return disabled || event.body === commandPrefix + 'disable';
+  // Add disabled flag for thread if it doesn't already exists
+  if(!isThreadDisabled.hasOwnProperty(event.thread_id)) {
+    isThreadDisabled[event.thread_id] = false;
+  }
+
+  if(event.body === commandPrefix + commands[0]) {
+    msgIndexEnable = 0;
+    msgIndexDisable = 1;
+    return true;
+  } else if (!isThreadDisabled[event.thread_id] && event.body.startsWith(commandPrefix)) { // Avoids counting if already disabled
+    counter += Date.now() - prevTimeStamp <= 1000 ? 1 : 0;
+    prevTimeStamp = Date.now();
+
+    possibleSpam = counter > counterLimit;
+    if(possibleSpam) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Keyword "if" must be followed by whitespace.](https://www.codacy.com/app/theprojectshop/Kassy/file/2554636148/issues/source?bid=3282987&fileBranchId=3282995#l47)